prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirTransformerTests.kt

```diff
@@ -332,42 +352,222 @@ class FhirTransformerTests {
-        patient.id = "def456"
-        bundle.addEntry().resource = patient
-        val transformer = FhirTransformer(FhirTransformSchema())
 
-        // Incompatible value types
+        // Can't currently create entry on the fly
         assertThat {
             transformer.setBundleProperty(
-                "Bundle.entry.resource.ofType(Patient).name.text", CodeableConcept(),
+                "Bundle.entry.resource.ofType(DiagnosticReport).status", CodeType("final"),
                 CustomContext(bundle, bundle), bundle, bundle
             )
-        }.isFailure()
+        }.isFailure().hasMessage("Can't add missing entry.")
+
+        val patient = Patient()
+        patient.id = "def456"
+        bundle.addEntry().resource = patient
+
+        // Can't currently create new resources on the fly
         assertThat {
             transformer.setBundleProperty(
-                "Bundle.entry.resource.ofType(Patient).active", StringType("nonBoolean"),
+                "Bundle.entry.resource.ofType(DiagnosticReport).status", CodeType("final"),
                 CustomContext(bundle, bundle), bundle, bundle
             )
         }.isFailure()
 
-        // Can't currently create new resources on the fly
+        // Improper extension format
+        transformer.setBundleProperty(
+            "Bundle.entry.resource.ofType(Patient).extension(regexNonMatch).value[x]", IdType("newId"),
+            CustomContext(bundle, bundle), bundle, bundle
+        )
+        verify(exactly = 1) { logger.warn(match<String> { it.contains("Could not find property") }) }
+
+        // Invalid bundleProperties
+        transformer.setBundleProperty(
+            "", IdType("newId"),
+            CustomContext(bundle, bundle), bundle, bundle
+        )
+        verify(exactly = 1) { logger.warn(match<String> { it.contains("bundleProperty was not set") }) }
+
+        transformer.setBundleProperty(
+            "id", IdType("newId"),
+            CustomContext(bundle, bundle), bundle, bundle
+        )
+        verify(exactly = 1) { logger.warn(match<String> { it.contains("Expected at least 2 parts") }) }
+
+        transformer.setBundleProperty(
+            "%key.text", StringType("SomeName"),
+            CustomContext(
+                bundle,
+                bundle,
+                constants = mutableMapOf(Pair("key", "Bundle.entry.resource.ofType(Patient).name"))
+            ),
+            bundle, bundle
+        )
+        verify(exactly = 1) {
+            logger.warn(match<String> { it.contains("cannot dynamically create components relying on constants") })
+        }
+
+        transformer.setBundleProperty(
+            "Bundle.entry.resource.ofType(Patient).name.%key", StringType("SomeName"),
+            CustomContext(bundle, bundle, constants = mutableMapOf(Pair("key", "text"))), bundle, bundle
+        )
+        verify(exactly = 1) { logger.warn(match<String> { it.contains("Constants not supported in lowest level") }) }
```

Comment:
Your test are verifying logs and not results. It makes it seem there should be some return to this functions and perhaps needs a refactor.